### DISCUSSION TITLE
How to create a Notion-like Toggle Plugin?

### DISCUSSION THREAD

**Discussion by gutenye** [facebook/lexical#1835] [Q&A]
Like below. Would love to see an example made into Playground. Thanks

https://user-images.githubusercontent.com/377544/163746866-7f479034-e675-42d9-98ab-6f20b0c9a3b9.mp4

**Comment by zurfyx** [comment]
That's an interesting one! The implementation is really attached to its requirements. For example, can it have siblings? How should text deletion work?

In the case of Notion it seems like it it behaves very similarly to ListNode so I can see two approaches:

## ElementNode

The most accessible and cross-front-end framework approach is to write your own ElementNode that displays the arrow, has some padding and some TextNode inside it.

The ElementNode afterInsert will render these hidden item nodes, and the arrow click handler will change their visibility (rather than create-delete that would not show in the EditorState and would be bad for export and collab)

## DecoratorNode

DecoratorNode is not necessarily a bad approach is there's lots of custom CSS involved. In the case the DecoratorNode would handle the handle the first line and the arrow and would render another editor inside it.

---

Opened https://github.com/facebook/lexical/issues/1841
